src/Features/Core/Portable/UnusedReferences/IReferenceCleanupService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.UnusedReferences
+{
+    internal interface IReferenceCleanupService : IWorkspaceService
+    {
+        /// <summary>
+        /// Gets the current selected TargetFrameworkMoniker for the specified project.
+        /// </summary>
+        string GetTargetFrameworkMoniker(ProjectId projectId);
+
+        /// <summary>
+        /// For the given project, returns the full path to the project.assets.json file
```

Comment:
A single project.assets.json is generated containing the dependency information for all TFMs.